addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4672,6 +4672,95 @@ registry.ReplaceMedia = SnippetOptionWidget.extend({
         this.$target.dblclick();
     },
 });
+ 
+/**
+ * Allows the image to redirect to custom URL on click.
+ */
+registry.LinkTool = SnippetOptionWidget.extend({
+
+    /**
+     * @override
+     */
+    start() {
+        this.linkEl = this.$target[0].parentElement.tagName === 'A' ?
+            this.$target[0].parentElement : null;
+        this.isNewWindow = this.linkEl ?
+            this.linkEl.getAttribute('target') === '_blank' : false;
```

Comment:
You don't need these variables. Your database is the DOM, the editor options values and visibility should only be reflected by the `_computeWidgetState` and `_computeWidgetVisibility` methods. It's easier not to keep a duplicated tracking of those values... simply read the DOM when you need them.